src/flb_log.c

```diff
@@ -121,7 +126,7 @@ static inline int log_read(flb_pipefd_t fd, struct flb_log *log)
     bytes = flb_pipe_read_all(fd, &msg, sizeof(struct log_message));
 
     if (bytes <= 0) {
-        flb_errno();
+        flb_pipe_error();
```

Comment:
In this case there is an `flb_error` call in the second error exitt code path inside of `flb_pipe_read_all` which could cause confusion, could you remove it?

Reply:
Good catch, fixed in [7a1429c](https://github.com/fluent/fluent-bit/pull/10017/commits/7a1429c720b7ac8aa9d74943b44be7c2743678cf)